fast_llm/data/data/gpt/data.py

```diff
@@ -77,18 +79,43 @@ def __init__(
         self._max_sequence_length = max_sequence_length
         self._cross_document_attention = cross_document_attention
 
+        # This is a mapping between the dataset name in the config and its normalized variant for later reference.
+        # This is needed because previously, dataset names were based on phases and were capitalized,
+        # but this is no longer a requirement.
+        self._dataset_name_map = {key.lower(): key for key in self._config.datasets.keys()}
+
     def setup(
         self,
         distributed: "Distributed",
-        samples_per_phase: dict[PhaseType, int],
+        samples_per_dataset: dict[str, int],
         cache_directory: pathlib.Path,
         timeout: float | None = None,
     ) -> None:
         """
         Load the datasets, and prepare or load the samplings.
         This may take a while and a significant amount of cpu memory.
         """
-        super().setup(distributed, samples_per_phase, cache_directory)
+        # Some dataset names may come from phases and are capitalized,
+        # so we need to normalize them before use.
+        samples_per_dataset = {key.lower(): value for key, value in samples_per_dataset.items()}
+
+        # Check and raise an error if a used dataset is not defined.
+        for dataset_name in samples_per_dataset.keys():
```

Comment:
If we want to do these check during validation, we could also do the check in the training config, which has access to both the data and evaluation configs.